tests/core/runners/test_ase.py

```diff
@@ -47,6 +52,196 @@ def teardown_function():
+        "max_micro_iterations",
+        "morse_scaling",
+        "geometry_friction",
+        "distance_cutoff",
+        "save_raw_path",
+        "expected_length",
+    ),
+    [
+        (20, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 20),  # Default parameters
+        (10, 2e-3, 15, 20, 1.7, 1e-2, 3.0, None, 10),  # Different number of images
+        (
+            20,
+            1e-4,
+            10,
+            10,
+            1.5,
+            0.01,
+            2.5,
+            "raw_path.xyz",
+            20,
+        ),  # Different interpolation parameters and save_raw
+    ],
+)
+def test_geodesic_interpolate_wrapper(
+    setup_test_environment,
+    nimages,
+    convergence_tolerance,
+    max_iterations,
+    max_micro_iterations,
+    morse_scaling,
+    geometry_friction,
+    distance_cutoff,
+    save_raw_path,
+    expected_length,
+):
+    reactant, product = setup_test_environment
+    # Execute the geodesic_interpolate_wrapper function
+    smoother_path = _geodesic_interpolate_wrapper(
+        reactant,
+        product,
+        nimages=nimages,
+        convergence_tolerance=convergence_tolerance,
+        max_iterations=max_iterations,
+        max_micro_iterations=max_micro_iterations,
+        morse_scaling=morse_scaling,
+        geometry_friction=geometry_friction,
+        distance_cutoff=distance_cutoff,
+    )
+    assert smoother_path[1].positions[0][0] == pytest.approx(1.36055556030, abs=1e-1)
+
+
+def test_geodesic_interpolate_wrapper_large_system(setup_test_environment):
+    rng = np.random.default_rng()  # Create a random number generator instance
+    large_atoms = Atoms("H" * 40, positions=rng.random((40, 3)))
+
+    # Test with large system to trigger sweeping updates
+    smoother_path = _geodesic_interpolate_wrapper(large_atoms, large_atoms)
+    assert len(smoother_path) == 20
+
+
+@pytest.mark.parametrize(
+    (
+        "method",
+        "optimizer_class",
+        "precon",
+        "n_intermediate",
+        "k",
+        "max_steps",
+        "fmax",
+        "expected_logfile",
+        "r_positions",
+        "p_energy",
+        "first_image_forces",
+        "second_images_positions",
+        "index_ts",
+        "pot_energy_ts",
+        "forces_ts",
+        "last_images_positions",
+    ),
+    [
+        # ("aseneb", NEBOptimizer, None, 10, 0.1, 3, 1e-3, None),
+        # ("aseneb", SciPyFminBFGS, None, 1000, 0.1, 3, 1e-3, "some_logdir",
+        #   0.78503956131, -24.9895786292, -0.0017252843, 0.78017739462, 9, -19.946616164,
+        #   -0.19927549, 0.51475535802),
+        (
+            "aseneb",
+            NEBOptimizer,
+            None,
+            10,
+            0.1,
+            3,
+            1e-3,
+            "some_logdir",
+            -0.854,
+            1.082,
+            -0.005,
+            -0.8161139,
+            9,
+            -19.946616164,
+            -0.19927549,
+            0.51475535802,
+        )
+    ],
+)
+def test_run_neb(
```

Comment:
Your test of `run_neb` should not rely on `geodesic-interpolate`. If you want to test the NEB method, generate the images using ASE's methods. This will ensure we are only testing one thing at a time (hence the name a "unit test").

Reply:
The NEB function needs list of ASE images. I am avoiding it because it will just become a long list of coordinates in this file. Not sure how to do it cleanly.